Chat: describe Message Streaming demo (#33419)
DevExpress/DevExtreme: apps/demos/Demos/Chat/MessageStreaming/description.md

> This demo integrates the DevExtreme Chat component with [Azure OpenAI](https://azure.microsoft.com/en-us/products/ai-services/openai-service) to stream AI-generated responses in real time. As the AI model produces output token by token, the Chat component renders the response incrementally inside a message bubble. A typing indicator appears while the response is being streamed, and the send button transforms into a stop button so users can cancel an in-progress stream at any time.

Reviewer: - I read "integrates" and expected the Chat to be a wrapper :) That's why I prefer to change the verb.
- "is being" is grammatically correct but I have  some doubts... Let's make an experiment.

```suggestion
This demo implements the DevExtreme Chat component with [Azure OpenAI](https://azure.microsoft.com/en-us/products/ai-services/openai-service) to stream AI-generated responses in real time. As the AI model produces output token by token, the Chat component renders the response incrementally inside a message bubble. A typing indicator appears while the response is being streamed, and the send button transforms into a stop button so that users can cancel an in-progress stream at any time.
```

> The empty state displays custom suggestion cards. Clicking a card sends the corresponding prompt directly to the AI without extra user input.

Reviewer: ```suggestion
The empty Chat displays custom suggestion cards. Clicking a card sends the corresponding prompt directly to the AI without extra user input.
```

<!--split-->

## Streaming AI Responses

> The demo calls the Azure OpenAI Chat Completions API with `stream: true`. Incoming delta chunks are passed through a `createDelayedRenderer` queue that introduces a short display delay between chunks to produce a smooth typing effect. Each chunk is appended to a growing buffer, and the assistant message is updated in the data store with every render cycle via a `dataSource.store().push(...)` update operation.

Reviewer: Can we somehow reduce the number of passives in the last sentence?

## Stopping a Stream

> An `AbortController` is created before each request and its signal is forwarded to the Azure OpenAI SDK. When the user clicks the stop button, `abortController.abort()` is called, which cancels the in-progress HTTP request.

Reviewer: I think we can reduce the number of passives here too.

> The demo creates an `AbortController` before each request and forwards its signal to the Azure OpenAI SDK. When the user clicks the stop button, the demo calls `abortController.abort()`, which cancels the in-progress HTTP request.

Reviewer: Please check technical correctness.

```suggestion
The demo creates an `AbortController` before each request and forwards its output to the Azure OpenAI SDK. When the user clicks the stop button, the demo calls `abortController.abort()` to cancel the in-progress HTTP request.
```

> The [sendButtonOptions](/Documentation/ApiReference/UI_Components/dxChat/Configuration/#sendButtonOptions) property switches the button's `action` to `'custom'` and its icon to `'stopfilled'` while streaming is active, then reverts to the default send configuration once streaming ends.

Reviewer: ```suggestion
The [sendButtonOptions](/Documentation/ApiReference/UI_Components/dxChat/Configuration/#sendButtonOptions) property switches the button's `action` property to `'custom'` and the icon to `'stopfilled'` while streaming is active, then reverts to the default (send) configuration once streaming ends.
```

## Custom Empty View

> The Chat component specifies an [emptyViewTemplate](/Documentation/ApiReference/UI_Components/dxChat/Configuration/#emptyViewTemplate) that replaces the default empty state with custom suggestion cards. Clicking a card constructs a message and triggers the demo's message-sending flow directly, bypassing the text input.

Reviewer: ```suggestion
The Chat component specifies an [emptyViewTemplate](/Documentation/ApiReference/UI_Components/dxChat/Configuration/#emptyViewTemplate) that replaces the default empty state with custom suggestion cards. Clicking a card constructs a message and triggers the demo message sending flow directly, bypassing the text input.
```

> The Chat component specifies an [emptyViewTemplate](/Documentation/ApiReference/UI_Components/dxChat/Configuration/#emptyViewTemplate) that replaces the default empty state with custom suggestion cards. Clicking a card constructs a message and triggers the demo message sending flow directly, bypassing the text input.

Reviewer: ```suggestion
The Chat component specifies an [emptyViewTemplate](/Documentation/ApiReference/UI_Components/dxChat/Configuration/#emptyViewTemplate) that replaces the default empty state with custom suggestion cards. Clicking a card creates a message and triggers the demo message sending flow directly, bypassing the text input.
```
